src/backend/cuda/kernel/random.hpp

```diff
@@ -142,14 +578,29 @@ namespace kernel
     }
 
     template<typename T>
-    void randu(T *out, size_t elements)
+    void randu(T *out, size_t elements, const af::randomType &rtype)
     {
-        int device = getActiveDeviceId();
-
-        int threads = THREADS;
-        int blocks  = divup(elements, THREADS);
-        if (blocks > BLOCKS) blocks = BLOCKS;
-        CUDA_LAUNCH(uniform_kernel, blocks, threads, out, states[device], elements);
+        switch (rtype) {
+            case AF_RANDOM_DEFAULT:
+                {
+                    int device = getActiveDeviceId();
+                    int threads = THREADS;
+                    int blocks  = divup(elements, THREADS);
+                    if (blocks > BLOCKS) blocks = BLOCKS;
+                    CUDA_LAUNCH(uniform_kernel, blocks, threads, out, states[device], elements);
+                    break;
+                }
+            case AF_RANDOM_PHILOX:
+                {
+                    int threads = THREADS;
+                    int elementsPerBlockIteration = THREADS*4*sizeof(unsigned)/sizeof(T);
+                    int blocks = divup(elements, elementsPerBlockIteration);
+                    CUDA_LAUNCH(philox_uniform_kernel, blocks, threads,
+                            out, seed, elementsPerBlockIteration, elements);
+                    ++seed;
+                    break;
+                }
+        }
```

Comment:
Also add the proper licensing for random_philox.hpp like we did for random.hpp in OpenCL backend.